nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -360,6 +361,15 @@ in
         '';
       };
 
+      systemFeatures = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = [ "kvm" "big-parallel" "gccarch-skylake" ];
```

Comment:
If this is used often, we should probably get nix to autodetect it

Reply:
I’m not sure if gccarch is the right name though. It really doesn’t have anything to do with gcc, it’s just what features the cpu supports.

Reply:
Yes, I was thinking about change the names to something like "can-build-for-skylake" or "can-run-skylake-code", but that would cause massive rebuild :)

Reply:
can-build-for-skylake should be every x86_64. gcc can output any arch

Reply:
Yeah, if it was up to me I would do ```intel-skylake```, ```intel-haswell```, etc.

Reply:
The ```default``` case from here
https://github.com/NixOS/nixpkgs/blob/75920367737e32c1aa70da06f3108b3af4322dd7/nixos/modules/services/misc/nix-daemon.nix#L491-L495
could be ```[ "gccarch-${pkgs.hostPlatform.platform.gcc.arch}" ]``` to support unlisted architectures (assuming they can run their own code).
adding ```intel-``` or ```amd-``` prefix would make in impossible

Reply:
Yeah prefixes aren't that useful. Using gcc's name make sense. Could this support ARM as well?